scripts.py

```diff
@@ -1493,9 +1493,8 @@ def work_from_metadata(self, collection, collection_type, metadata, policy, *arg
 
         edition, new = metadata.edition(self._db)
         metadata.apply(edition, collection, replace=policy)
-        data_source = metadata.data_source(self._db)
         [pool] = [x for x in edition.license_pools
-                  if x.data_source == data_source]
+                  if x.collection == collection]
```

Comment:
`Edition.license_pools` already filters license pools with the edition's data source (and primary identifier), but not by collection. So, we need to ensure that we have the right one for our collection.